<a href="https://colab.research.google.com/github/gtoubian/cce/blob/main/Project_3_Gabriel_Toubian.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Project 3: NHL Player Salaries**
**By: Gabriel Toubian**

This project will involve the yearly salaries for NHL players and the different factors that influence them. We can make inferences that determine what makes a player more "valuable" to an NHL team (earn a higher salary).

**Importing the dataset and building a pandas dataframe out of it**

In [3]:
from google.colab import files
data = files.upload()

Saving nhl.csv to nhl (1).csv


In [7]:
import pandas as pd

import io
nhl = pd.read_csv('nhl.csv', encoding = "ISO-8859-1")

In [10]:
nhl = nhl.dropna(how='any')

In [14]:
nhl.tail()

Salary      Born        City Pr/St Cntry  ... Grit  DAP   Pace    GS  GS/G
606   680000  93-04-16  Burnsville    MN   USA  ...   49  8.7  107.1  -1.2 -0.09
607   600000  92-09-26      Denver    CO   USA  ...  100  6.9  103.6  23.2  0.33
608  9000000  85-01-21     Madison    WI   USA  ...  213  6.3  111.3  44.1  0.54
609  4250000  87-03-30    Montreal    QC   CAN  ...  200  4.1  108.8  32.5  0.43
611   925000  93-05-12  St. Albert    AB   CAN  ...  229  7.4  104.0  45.8  0.56

[5 rows x 154 columns]

We can see this dataset includes the salaries and stats for 611 players.

# Hypothesis testing

Whether a player is left or right handed can have on impact on the type of position they can play optimally. Therefore, do players of a certain handedness come at a premium? Let's start by seeing if whether a player is **left or right handed** has an impact on their salary. We can run a one-way ANOVA test to see if this factor is statistically significant.

In [20]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
import numpy as np
import scipy.stats as sc

In [22]:
nhlhand = nhl[["Hand", "Salary"]]
Left = np.array(nhlhand[nhlhand["Hand"]=="L"]["Salary"])
Right = np.array(nhlhand[nhlhand["Hand"]=="R"]["Salary"])

sc.f_oneway(Left, Right)

F_onewayResult(statistic=0.03681398777778426, pvalue=0.8479533459419761)

With a p-value of well over 0.05, we cannot reject the null hypothesis, which means that a player's handedness doesn't necessarily play a sigificant part in determining their salary.

How about a player's **Nationality**? Do certain countries have a tendency of producing more "valuable" players through their programs? Let's take 2 countries that have a tendency to finish relatively high in country-based competitions: **Canada and USA**

In [58]:
nhlnat = nhl[["Nat", "Salary"]]

In [60]:
nhlnat = nhlnat[(nhlnat.Nat == 'CAN') | (nhlnat.Nat == 'USA')]

In [63]:
Canadian = np.array(nhlnat[nhlnat["Nat"]=="CAN"]["Salary"])
American = np.array(nhlnat[nhlnat["Nat"]=="USA"]["Salary"])

sc.f_oneway(Canadian, American)

F_onewayResult(statistic=0.019945699195699655, pvalue=0.8877687260535069)

With a p-value of well over 0.05, we cannot reject the null hypothesis, which means that a player's nationality (in this case, whether they are Canadian or American) doesn't necessarily play a sigificant part in determining their salary.

# Multiple Regression

In [71]:
nhl2=nhl.copy()

In [72]:
del nhl2['Salary']
del nhl2['Born']
del nhl2['City']
del nhl2['Pr/St']
del nhl2['Cntry']
del nhl2['Nat']
del nhl2['DftYr']
del nhl2['Hand']
del nhl2['Last Name']
del nhl2['First Name']
del nhl2['Position']
del nhl2['Team']

In [74]:
X = nhl2
y = nhl['Salary']
sm.OLS(y, X).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                 Salary   R-squared (uncentered):                   0.904
Model:                            OLS   Adj. R-squared (uncentered):              0.848
Method:                 Least Squares   F-statistic:                              16.01
Date:                Fri, 23 Apr 2021   Prob (F-statistic):                    7.72e-69
Time:                        16:44:00   Log-Likelihood:                         -5493.7
No. Observations:                 359   AIC:                                  1.125e+04
Df Residuals:                     226   BIC:                                  1.177e+04
Df Model:                         133                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Ht         -1.049e+05   6.03e+04     -1.740      0.083   -2.24e+05    1.39e+04
Wt          2.711e+04   9264.395      2.926      0.004    8856.513    4.54e+04
DftRd      -1.762e+05   3.08e+05     -0.572      0.568   -7.83e+05    4.31e+05
Ovrl        4083.7337   9921.463      0.412      0.681   -1.55e+04    2.36e+04
GP         -7.935e+04   9.09e+04     -0.873      0.384   -2.58e+05    9.98e+04
G          -2.674e+06   3.09e+06     -0.865      0.388   -8.76e+06    3.41e+06
A          -9.474e+06   2.78e+06     -3.412      0.001   -1.49e+07      -4e+06
A1          5.755e+06   2.17e+06      2.657      0.008    1.49e+06       1e+07
A2          5.715e+06   2.16e+06      2.645      0.009    1.46e+06    9.97e+06
PTS         3.899e+06   1.92e+06      2.032      0.043    1.19e+05    7.68e+06
+/-        -2902.2398    3.1e+04     -0.094      0.926    -6.4e+04    5.82e+04
E+/-        4.434e+04   4.79e+04      0.925      0.356   -5.01e+04    1.39e+05
PIM         1.181e+04    9.6e+04      0.123      0.902   -1.77e+05    2.01e+05
Shifts     -3462.9948   1476.993     -2.345      0.020   -6373.433    -552.557
TOI         -417.0925    556.409     -0.750      0.454   -1513.506     679.321
TOIX        3.197e+04   3.33e+04      0.960      0.338   -3.37e+04    9.76e+04
TOI/GP      1.198e+06   2.34e+06      0.513      0.609    -3.4e+06     5.8e+06
TOI/GP.1    -8.99e+05   2.34e+06     -0.384      0.702   -5.52e+06    3.72e+06
TOI%       -1.628e+05   1.69e+05     -0.966      0.335   -4.95e+05    1.69e+05
IPP%       -4034.9291   5179.287     -0.779      0.437   -1.42e+04    6170.940
SH%         7.482e+05    1.9e+06      0.394      0.694   -2.99e+06    4.49e+06
SV%         7.055e+07   1.89e+08      0.374      0.709   -3.01e+08    4.42e+08
PDO        -6.772e+04   1.89e+05     -0.359      0.720    -4.4e+05    3.04e+05
F/60       -2.205e+07    1.8e+07     -1.223      0.223   -5.76e+07    1.35e+07
A/60        2.186e+07    1.8e+07      1.212      0.227   -1.37e+07    5.74e+07
Pct%       -6336.6469   9267.814     -0.684      0.495   -2.46e+04    1.19e+04
Diff       -1.647e+04   1.78e+04     -0.926      0.355   -5.15e+04    1.86e+04
Diff/60     2.184e+07    1.8e+07      1.210      0.227   -1.37e+07    5.74e+07
iCF        -1.834e+05    7.5e+04     -2.444      0.015   -3.31e+05   -3.55e+04
iCF.1       1.866e+05   7.65e+04      2.440      0.015    3.59e+04    3.37e+05
iFF         7.268e+05   3.29e+05      2.211      0.028    7.92e+04    1.37e+06
iSF        -1.037e+06   4.19e+05     -2.476      0.014   -1.86e+06   -2.12e+05
iSF.1       2.768e+05   8.36e+05      0.331      0.741   -1.37e+06    1.92e+06
iSF.2      -1.708e+06   1.97e+06     -0.866      0.388    -5.6e+06    2.18e+06
ixG        -1255.2677   2.74e+05     -0.005      0.996    -5.